src/components/Header/Header.js

```diff
@@ -1,11 +1,38 @@
+import React from "react";
 import "./Header.css";
 
-function Header() {
-  return (
-    <div className="header">
-      <h2 className="header-item">Header</h2>
-    </div>
-  );
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    }
+  }
+
+  onCheckboxUnderHeaderChange = () => {
+      this.props.onCheckboxUnderHeaderChange()
+  }; 
```

Comment:
No actual reason to declare a separate function. You can just pass it to event handler.